src/ZipkinBundle/Components/Messenger/ZipkinSendHandler.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+
+namespace ZipkinBundle\Components\Messenger;
+
+use Symfony\Component\Messenger\Envelope;
+
+use Zipkin\Kind;
+use Zipkin\Propagation\B3;
+use Zipkin\Tags;
+use Zipkin\Tracing;
+
+class ZipkinSendHandler
+{
+    /**
+     * @var \Zipkin\Tracer
+     */
+    private $tracer;
+    /**
+     * @var callable
+     */
+    private $injector;
+
+    public function __construct(Tracing $tracing)
+    {
+        $this->tracer = $tracing->getTracer();
+        $this->injector = $tracing->getPropagation()->getInjector(new PropagationStamp());
+    }
+
+    public function handle(Envelope $envelope): Envelope
+    {
+        if ($envelope->all(B3Stamp::class)) {
+            return $envelope;
+        }
+
+        $span = $this->tracer->nextSpan();
```

Comment:
Not sure if within this library patterns like `consume->process->send` are possible. If they are, incoming request can already contains a trace context. 

reference: 

https://github.com/openzipkin/brave/blob/b8ee69ef7a463a7ed364374536b2645529d508b9/instrumentation/kafka-clients/src/main/java/brave/kafka/clients/TracingProducer.java#L99-L112